Decorator/ContentTypeProviderDetachDecorator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+* This file is part of the Integrated package.
+*
+* (c) e-Active B.V. <integrated@e-active.nl>
+*
+* For the full copyright and license information, please view the LICENSE
+* file that was distributed with this source code.
+*/
+
+namespace Integrated\Bundle\SolrBundle\Decorator;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Integrated\Common\Solr\Task\Provider\ContentTypeProviderInterface;
+use Integrated\Bundle\SolrBundle\Iterator\DetachIterator;
+
+/**
+ * @author Patrick Mestebeld <patrick@e-active.nl>
+ */
+class ContentTypeProviderDetachDecorator implements ContentTypeProviderInterface
+{
+    /**
+     * @var ContentTypeProviderInterface
+     */
+    private $provider;
+
+    /**
+     * @var ObjectManager
+     */
+    private $manager;
+
+    /**
+     * @param ContentTypeProviderInterface $provider
+     * @param ObjectManager $manager
+     */
+    public function __construct(ContentTypeProviderInterface $provider, ObjectManager $manager)
+    {
+        $this->provider = $provider;
+        $this->manager = $manager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getContent($id)
+    {
+        return new DetachIterator($this->provider->getContent($id), $this->manager);
```

Comment:
ContentTypeProviderInterface does not return an Iterator according to the docblock. The best way to fix is is to change the docblock (change Traversable into Iterator) of the ContentTypeProviderInterface (which is in the library)